internal/service/ec2/vpnsite_connection.go

```diff
@@ -960,6 +1024,10 @@ func expandVPNTunnelOptionsSpecification(d *schema.ResourceData, prefix string)
 		}
 	}
 
+	if v, ok := d.GetOk(prefix + "log_options"); ok && v.(*schema.Set).Len() > 0 {
+		apiObject.LogOptions = expandVPNTunnelLogOptionsSpecification(d)
+	}
```

Comment:
```suggestion
	if v, ok := d.GetOk(prefix + "log_options"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
		apiObject.LogOptions = expandVPNTunnelLogOptionsSpecification(v.([]interface{})[0].(map[string]interface{}))
	}
```

and change parameter of `expandVPNTunnelLogOptionsSpecification` to `tfMap map[string]interface{}`.

Similar on for _flatten_.